### DISCUSSION TITLE
Is there a way to read one query string param without serde at all?

### DISCUSSION THREAD

**Discussion by frederikhors** [tokio-rs/axum#1635] [Q&A] [answered]
Hi guys!

I read [this example code](https://github.com/tokio-rs/axum/blob/main/examples/query-params-with-empty-strings/src/main.rs) and I understand that `serde` is necessarily.

I have to detect only one param in query string: `https://domain.com/players/123456789`, the number: do you think there is a way to not add `serde` to my project at all?

**Comment by davidpdrsn** [comment, marked as answer]
Do you mean query params or path params? `https://domain.com/players/123456789` doesn't contain query params.

Regardless you can parse it as `Query<HashMap<String, String>>` or `Path<HashMap<String, String>>`.

For path params you can also use `Path<T>` for any `T: Deserialize`. For example `Path<i32>`
